src/modules/roc_pipeline/config.h

```diff
@@ -135,7 +135,7 @@ struct ReceiverConfig {
     bool timing;
 
     ReceiverConfig()
-        : sample_rate(DefaultSampleRate)
+        : sample_rate(0)
```

Comment:
Why this change? The idea of default constructors of these *Config structures is that if a field was not overwritten explicitly, it has a reasonable value. This feature will be used in roc_lib, which will overwrite only those fields that were set to non-zero by user.

Reply:
Providing zero sample rate for sox means, that it'll determine sample rate of the output file or device by its own.

UPD: Later I've checked, that if we provide unsupported sample rate to sox (e.g. 44K, while device supports 48K), sox will also detect the correct sample rate by its own.

I'll rollback this change.

Reply:
> This feature will be used in roc_lib, which will overwrite only those fields that were set to non-zero by user.

Where do we check that the user provides non-zero values?

```cpp
bool make_receiver_config(pipeline::ReceiverConfig& out, const roc_receiver_config* in) {
    out.default_session.latency = in->latency;
    out.default_session.timeout = in->timeout;
    out.default_session.samples_per_packet = in->samples_per_packet;
    out.sample_rate = in->sample_rate;
   
    //...
}
```

We'll panic only during a session creation.

Reply:
> Later I've checked, that if we provide unsupported sample rate to sox (e.g. 44K, while device supports 48K), sox will also detect the correct sample rate by its own.

I think we should not rely on this, it's too implicit and can be changed when switching from SoX to something else.

I suggest the following:

* Use a reasonable default in ReceiverConfig::sample_rate (44100). This default will be ignored in roc_recv, but it will be used in roc_lib.

* Use a separate variable in roc_recv, initialize it with `--rate` option value or zero if it wan't given, pass it to SoX, then get the actual sample rate from SoX, and write it to ReceiverConfig.

> Where do we check that the user provides non-zero values?

There are no checks yet. However, I plan to add them in 0.1. It will look like:

```
if (in->sample_rate) {
    out.sample_rate = in->sample_rate;
}
```

Reply:
Fixed